<a href="https://colab.research.google.com/github/shyamalkar/Data_analysis_python_07/blob/main/Data_analysis_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

***`Import CSV file`***

In [ ]:
import pandas as pd
import kagglehub

# Download latest version
path = ("/content/collegePlace.csv")

print("Path to dataset files:", path)





Path to dataset files: /content/collegePlace.csv


In [ ]:
df = pd.read_csv(path)
print(df)

      Age  Gender                         Stream  Internships  CGPA  Hostel  \
0      22    Male  Electronics And Communication            1     8       1   
1      21  Female               Computer Science            0     7       1   
2      22  Female         Information Technology            1     6       0   
3      21    Male         Information Technology            0     8       0   
4      22    Male                     Mechanical            0     8       1   
...   ...     ...                            ...          ...   ...     ...   
2961   23    Male         Information Technology            0     7       0   
2962   23    Male                     Mechanical            1     7       1   
2963   22    Male         Information Technology            1     7       0   
2964   22    Male               Computer Science            1     7       0   
2965   23    Male                          Civil            0     8       0   

      HistoryOfBacklogs  PlacedOrNot  
0           

**Let's use head function . (head) function help to see first 5 rows**


In [ ]:
df.head()

Age  Gender                         Stream  Internships  CGPA  Hostel  \
0   22    Male  Electronics And Communication            1     8       1   
1   21  Female               Computer Science            0     7       1   
2   22  Female         Information Technology            1     6       0   
3   21    Male         Information Technology            0     8       0   
4   22    Male                     Mechanical            0     8       1   

   HistoryOfBacklogs  PlacedOrNot  
0                  1            1  
1                  1            1  
2                  0            1  
3                  1            1  
4                  0            1

**Let's use tail() function . tail() usage for last 5 rows**

In [ ]:
df.tail(5)

Age Gender                  Stream  Internships  CGPA  Hostel  \
2961   23   Male  Information Technology            0     7       0   
2962   23   Male              Mechanical            1     7       1   
2963   22   Male  Information Technology            1     7       0   
2964   22   Male        Computer Science            1     7       0   
2965   23   Male                   Civil            0     8       0   

      HistoryOfBacklogs  PlacedOrNot  
2961                  0            0  
2962                  0            0  
2963                  0            0  
2964                  0            0  
2965                  0            1

**To see which type of data here ? So we use this dtypes. **

In [ ]:
df.dtypes

Age                   int64
Gender               object
Stream               object
Internships           int64
CGPA                  int64
Hostel                int64
HistoryOfBacklogs     int64
PlacedOrNot           int64
dtype: object

**Let's use describe() function to see data frame summary statistics, but describe () use only for numeric columns(int / float)**

In [ ]:
df.describe()

Age  Internships         CGPA       Hostel  HistoryOfBacklogs  \
count  2966.000000  2966.000000  2966.000000  2966.000000        2966.000000   
mean     21.485840     0.703641     7.073837     0.269049           0.192178   
std       1.324933     0.740197     0.967748     0.443540           0.394079   
min      19.000000     0.000000     5.000000     0.000000           0.000000   
25%      21.000000     0.000000     6.000000     0.000000           0.000000   
50%      21.000000     1.000000     7.000000     0.000000           0.000000   
75%      22.000000     1.000000     8.000000     1.000000           0.000000   
max      30.000000     3.000000     9.000000     1.000000           1.000000   

       PlacedOrNot  
count  2966.000000  
mean      0.552596  
std       0.497310  
min       0.000000  
25%       0.000000  
50%       1.000000  
75%       1.000000  
max       1.000000

"""
What is the means of this output, when we use describe(include="all") function.

count -> count mean not all age number sum , count means 2966 student's data row have, no missing value have inside this dataset.

mean -> mean means average age is 21.

std -> mean std means standard devision . if result is min that means not high age and low age .

min -> min means most small age is 19 .

max -> usage for high age 30

25% -> 25% means 25% people are under age , that mean 25 % STUDENT AEG < 21.

50% = MIDDLE AGE IS 21 . this is called median .

75% AGE = 22. , that means 75% student's age upper 22 or below 22 .


In Short :--
      
      Count           How many student

      mean            average

      std           How much is it spread?

      min           most small number

      25%           below 25%

      50%            middle

      75%            below 75%

      max            The biggest




"""

***Describe(include="all")
Every column give summary
numeric + categorical + object all .
all column summary.
***

In [ ]:
df.describe(include="all")

Age Gender            Stream  Internships         CGPA  \
count   2966.000000   2966              2966  2966.000000  2966.000000   
unique          NaN      2                 6          NaN          NaN   
top             NaN   Male  Computer Science          NaN          NaN   
freq            NaN   2475               776          NaN          NaN   
mean      21.485840    NaN               NaN     0.703641     7.073837   
std        1.324933    NaN               NaN     0.740197     0.967748   
min       19.000000    NaN               NaN     0.000000     5.000000   
25%       21.000000    NaN               NaN     0.000000     6.000000   
50%       21.000000    NaN               NaN     1.000000     7.000000   
75%       22.000000    NaN               NaN     1.000000     8.000000   
max       30.000000    NaN               NaN     3.000000     9.000000   

             Hostel  HistoryOfBacklogs  PlacedOrNot  
count   2966.000000        2966.000000  2966.000000  
unique          NaN                NaN          NaN  
top             NaN                NaN          NaN  
freq            NaN                NaN          NaN  
mean       0.269049           0.192178     0.552596  
std        0.443540           0.394079     0.497310  
min        0.000000           0.000000     0.000000  
25%        0.000000           0.000000     0.000000  
50%        0.000000           0.000000     1.000000  
75%        1.000000           0.000000     1.000000  
max        1.000000           1.000000     1.000000

"This df.info() usage for showing full structure.
          
          How many row have ?

          How many column have ?

          Which column have ?

          Where is the missing value ?

          How much memory need ?

          
"

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2966 entries, 0 to 2965
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Age                2966 non-null   int64 
 1   Gender             2966 non-null   object
 2   Stream             2966 non-null   object
 3   Internships        2966 non-null   int64 
 4   CGPA               2966 non-null   int64 
 5   Hostel             2966 non-null   int64 
 6   HistoryOfBacklogs  2966 non-null   int64 
 7   PlacedOrNot        2966 non-null   int64 
dtypes: int64(6), object(2)
memory usage: 185.5+ KB


***`What is the meaning of this output ? `***

      RangeIndex:-
               Total = 2966 row have in this dataset .
                and 0-2965 means if count 0 then it's count 2955 because it's start coutn 0 and both are same .

      Non-Null Count

               2966 non-null means no missing value have in this column. If mmissing value have then it's show a error:- 2800 non-null int64 .
                
               Another
               e.g.,
               Total Row = 4
               Non-null = 3

               then info() show:-
               4 entries
               Age 3 non-null float64